packages/contracts-periphery/contracts/foundry-tests/Drippie.t.sol

```diff
@@ -0,0 +1,565 @@
+     *         possible to execute a drip.
+     */
+    function _warpToExecutable(string memory name) internal {
+        Drippie.DripConfig memory config = drippie.dripConfig(name);
+        vm.warp(config.interval + drippie.dripStateLast(name));
+    }
+
+    /**
+     * @notice Creates a drip and asserts that it was configured as expected.
+     */
+    function test_create_success() external {
+        Drippie.DripConfig memory cfg = _defaultConfig();
+        vm.expectEmit(true, true, true, true);
+        emit DripCreated(dripName, dripName, cfg);
+
+        vm.prank(drippie.owner());
+        drippie.create(dripName, cfg);
+
+        Drippie.DripStatus status = drippie.dripStatus(dripName);
+        Drippie.DripConfig memory config = drippie.dripConfig(dripName);
+
+        assertEq(uint256(status), uint256(Drippie.DripStatus.PAUSED));
+
+        assertEq(config.interval, cfg.interval);
+        assertEq(config.reentrant, cfg.reentrant);
+        assertEq(address(config.dripcheck), address(cfg.dripcheck));
+        assertEq(config.checkparams, cfg.checkparams);
+
+        assertEq(config.actions.length, cfg.actions.length);
+
+        for (uint256 i; i < config.actions.length; i++) {
+            Drippie.DripAction memory a = config.actions[i];
+            Drippie.DripAction memory b = cfg.actions[i];
+
+            assertEq(a.target, b.target);
+            assertEq(a.data, b.data);
+            assertEq(a.value, b.value);
+        }
+    }
+
+    /**
+     * @notice Ensures that the same drip cannot be created two times.
+     */
+    function test_create_fails_twice() external {
+        vm.startPrank(drippie.owner());
+        Drippie.DripConfig memory cfg = _defaultConfig();
+        drippie.create(dripName, cfg);
+        vm.expectRevert("Drippie: drip with that name already exists");
+        drippie.create(dripName, cfg);
+        vm.stopPrank();
+    }
+
+    /**
+     * @notice Ensures that only the owner of Drippie can create a drip.
+     */
+    function test_fails_unauthorized() external {
+        vm.expectRevert("UNAUTHORIZED");
+        drippie.create(dripName, _defaultConfig());
+    }
+
+    /**
+     * @notice The owner should be able to set the status of the drip.
+     */
+    function test_set_status_success() external {
+        vm.expectEmit(true, true, true, true);
+        emit DripCreated(dripName, dripName, _defaultConfig());
+        _createDefaultDrip(dripName);
+
+        address owner = drippie.owner();
+
+        {
+            Drippie.DripStatus status = drippie.dripStatus(dripName);
+            assertEq(uint256(status), uint256(Drippie.DripStatus.PAUSED));
+        }
+
+        vm.prank(owner);
+
+        vm.expectEmit(true, true, true, true);
+        emit DripStatusUpdated({
+            nameref: dripName,
+            name: dripName,
+            status: Drippie.DripStatus.ACTIVE
+        });
+
+        drippie.status(dripName, Drippie.DripStatus.ACTIVE);
```

Comment:
I wish this method was `setStatus` as opposed to `status` to make it's state transition explicit.

Reply:
+1 it looks a lot like a getter not a setter

Reply:
Unfortunately we can't change the abi right now and drippie v1 works well enough for now, there are some concepts for v2 out there but no pressing need to implement right now besides for fun